R/plot_aggregate_loanbooks.R

```diff
@@ -218,56 +236,58 @@ plot_aggregate_loanbooks <- function(config) {
   region_scatter_alignment_exposure <- region_select
   currency <- unique(company_aggregated_alignment_net[["loan_size_outstanding_currency"]])
   if (length(by_group) <= 1) {
-    if (
-      nrow(loanbook_exposure_aggregated_alignment_net) > 0
-    ) {
-      data_scatter_alignment_exposure <- loanbook_exposure_aggregated_alignment_net %>%
-        prep_scatter_alignment_exposure(
-          year = year_scatter_alignment_exposure,
-          region = region_scatter_alignment_exposure,
-          scenario = scenario_select,
-          group_var = by_group,
-          exclude_groups = "benchmark"
-        )
+    if (!is.null(loanbook_exposure_aggregated_alignment_net)) {
```

Comment:
can these multi-level if statements be combined with `&` or `|`? seems overly complicated at the moment

Reply:
one would hope so, not totally sure how best to do it though. I am not sure testing for `is.null(df)` and `nrow(df) > 0` can be combined? At least I thought they could not. But to avoid gross repetition, I could define the conditions further above and re-use them where possible. Also I guess `length(by_group)` could possibly be combined with `nrow(df)`... will check

Reply:
df <- NULL

if (!is.null(df)) {
  if (
    nrow(df) > 0
  ) {
    print("df is not empty")
  } else {
    print("df is empty")
  }
} else {
  print("df is NULL")
}
#> [1] "df is NULL"

df <- tibble::tibble()

if (!is.null(df)) {
  if (
    nrow(df) > 0
  ) {
    print("df is not empty")
  } else {
    print("df is empty")
  }
} else {
  print("df is NULL")
}
#> [1] "df is empty"

df <- tibble::tibble(a = 1)

if (!is.null(df)) {
  if (
    nrow(df) > 0
  ) {
    print("df is not empty")
  } else {
    print("df is empty")
  }
} else {
  print("df is NULL")
}
#> [1] "df is not empty"

df <- NULL

!is.null(df) & nrow(df) > 0
#> logical(0)

if (!is.null(df) & nrow(df) > 0) {
  print("does this work?")
} else {
  print("does this work?")
}
#> Error in if (!is.null(df) & nrow(df) > 0) {: argument is of length zero
Created on 2024-10-30 with [reprex v2.1.0](https://reprex.tidyverse.org/)